server/src/main/java/org/opensearch/index/shard/CheckpointRefreshListener.java

```diff
@@ -40,7 +40,7 @@ public void beforeRefresh() throws IOException {
 
     @Override
     public void afterRefresh(boolean didRefresh) throws IOException {
-        if (didRefresh) {
+        if (didRefresh && shard.getReplicationTracker().isPrimaryMode()) {
             publisher.publish(shard);
         }
```

Comment:
Should we also reject checkpoints if the replica copy(where the checkpoint was sent to) is operating in primary mode.

Reply:
+1

Reply:
Also on that note - we will want to cancel any ongoing replication events on both sides.  I've added https://github.com/opensearch-project/OpenSearch/issues/4136 to cover this.

Reply:
is it also worth checking shardRouting here?

Reply:
Added Logic to reject checkpoints if shard is in PrimaryMode in latest commit.